[2022-07-02 12:28:47,465] {taskinstance.py:1160} INFO - Dependencies all met for <TaskInstance: process_flowers_dag._get_flowers_from_postgres manual__2022-07-02T12:28:46.308083+00:00 [queued]>
[2022-07-02 12:28:47,481] {taskinstance.py:1160} INFO - Dependencies all met for <TaskInstance: process_flowers_dag._get_flowers_from_postgres manual__2022-07-02T12:28:46.308083+00:00 [queued]>
[2022-07-02 12:28:47,481] {taskinstance.py:1357} INFO - 
--------------------------------------------------------------------------------
[2022-07-02 12:28:47,481] {taskinstance.py:1358} INFO - Starting attempt 1 of 1
[2022-07-02 12:28:47,481] {taskinstance.py:1359} INFO - 
--------------------------------------------------------------------------------
[2022-07-02 12:28:47,504] {taskinstance.py:1378} INFO - Executing <Task(PythonOperator): _get_flowers_from_postgres> on 2022-07-02 12:28:46.308083+00:00
[2022-07-02 12:28:47,511] {standard_task_runner.py:52} INFO - Started process 4000 to run task
[2022-07-02 12:28:47,514] {standard_task_runner.py:79} INFO - Running: ['***', 'tasks', 'run', 'process_flowers_dag', '_get_flowers_from_postgres', 'manual__2022-07-02T12:28:46.308083+00:00', '--job-id', '50', '--raw', '--subdir', 'DAGS_FOLDER/process_iris_data.py', '--cfg-path', '/tmp/tmp97a7ca9m', '--error-file', '/tmp/tmp3ev_j56_']
[2022-07-02 12:28:47,515] {standard_task_runner.py:80} INFO - Job 50: Subtask _get_flowers_from_postgres
[2022-07-02 12:28:47,593] {task_command.py:370} INFO - Running <TaskInstance: process_flowers_dag._get_flowers_from_postgres manual__2022-07-02T12:28:46.308083+00:00 [running]> on host a5e5aec483b5
[2022-07-02 12:28:47,748] {taskinstance.py:1572} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=***
AIRFLOW_CTX_DAG_ID=process_flowers_dag
AIRFLOW_CTX_TASK_ID=_get_flowers_from_postgres
AIRFLOW_CTX_EXECUTION_DATE=2022-07-02T12:28:46.308083+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=manual__2022-07-02T12:28:46.308083+00:00
[2022-07-02 12:28:47,764] {base.py:68} INFO - Using connection ID 'postgres_conn' for task execution.
[2022-07-02 12:28:47,771] {python.py:173} INFO - Done. Returned value was: [{'id': 1, 'sepal_length': Decimal('5.1'), 'sepal_width': Decimal('3.5'), 'petal_length': Decimal('1.4'), 'petal_width': Decimal('0.2'), 'variety': 'Setosa'}, {'id': 2, 'sepal_length': Decimal('4.9'), 'sepal_width': Decimal('3'), 'petal_length': Decimal('1.4'), 'petal_width': Decimal('0.2'), 'variety': 'Setosa'}, {'id': 3, 'sepal_length': Decimal('4.7'), 'sepal_width': Decimal('3.2'), 'petal_length': Decimal('1.3'), 'petal_width': Decimal('0.2'), 'variety': 'Setosa'}]
[2022-07-02 12:28:47,781] {xcom.py:585} ERROR - Could not serialize the XCom value into JSON. If you are using pickle instead of JSON for XCom, then you need to enable pickle support for XCom in your *** config.
[2022-07-02 12:28:47,782] {taskinstance.py:1890} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/utils/session.py", line 68, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 2395, in xcom_push
    session=session,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/utils/session.py", line 68, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/xcom.py", line 197, in set
    map_index=map_index,
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/xcom.py", line 582, in serialize_value
    return json.dumps(value).encode('UTF-8')
  File "/usr/local/lib/python3.7/json/__init__.py", line 231, in dumps
    return _default_encoder.encode(obj)
  File "/usr/local/lib/python3.7/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/local/lib/python3.7/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/usr/local/lib/python3.7/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type Decimal is not JSON serializable
[2022-07-02 12:28:47,795] {taskinstance.py:1401} INFO - Marking task as FAILED. dag_id=process_flowers_dag, task_id=_get_flowers_from_postgres, execution_date=20220702T122846, start_date=20220702T122847, end_date=20220702T122847
[2022-07-02 12:28:47,813] {standard_task_runner.py:97} ERROR - Failed to execute job 50 for task _get_flowers_from_postgres (Object of type Decimal is not JSON serializable; 4000)
[2022-07-02 12:28:47,847] {local_task_job.py:156} INFO - Task exited with return code 1
[2022-07-02 12:28:47,892] {local_task_job.py:273} INFO - 0 downstream tasks scheduled from follow-on schedule check
